supervised/preprocessing/eda.py

```diff
@@ -211,24 +207,38 @@ def extensive_eda(df, y, save_path):
                             weight="bold",
                             alpha=0.75,
                         )
-                        plt.savefig(os.path.join(save_path, f"{col}_target"))
+                        plt.savefig(
+                            os.path.join(
+                                save_path,
+                                "{}_target".format(re.sub(r'[\\/*?:"<>|]', "_", col)),
+                            )
+                        )
 
                     elif PreprocessingUtils.get_type(X[col]) in (
                         "categorical",
                         "discrete",
                     ):
 
-                        if np.nunique(X[col]) <= 7:
+                        if X[col].nunique() > 7:
+                            warnings.warn("Considering 7 the most frequent values")
```

Comment:
Can we print only one warning for the whole dataset?

Reply:
There will be only one with this code. Only one of them will be executed depending on the type of `target`